Legacy/bonej/src/main/java/org/bonej/plugins/ParticleAnalysis.java

```diff
@@ -749,10 +756,21 @@ private static double getSurfaceArea(final List<Point3f> points) {
 						itb = surface.listIterator(ita.nextIndex());
 						while (itb.hasNext()) {
 							b = itb.next();
-							feret = Math.max(feret, a.distance(b));
+							final double distance = a.distance(b);
+							if (distance > feret) {
+								feret = distance;
+								feretA = a;
+								feretB = b;
+							}
```

Comment:
Since the list is not mutated, this can be a simple nested loop:
```
double feret = 0;
Point3f feretA = new Point3f();
Point3f feretB = new Point3f();
for (int p = 0; p < surface.size() - 1; p++) {
  final Point3f a = surface.get(p);
  for (int q = p + 1; q < surface.size(); q++) {
    final Point3f b = surface.get(q);
    final double distance = a.distance(b);
    if (distance > feret) {
      feret = distance;
      feretA = a;
      feretB = b;
    }
  }
}
```

Reply:
Btw, what should happen if there's only one point in the list? `feret = 0`?

Reply:
Good question, but it shouldn't happen because the point lists are output from the surface mesher which is putting points in in groups of 3. The smallest should be a tetrahedron, so 4 triangles and 12 points. Empty surfaces (nPoints = 0) are already handled.

Reply:
I think that assumption should be documented and checked for:
```
@param surfacePoints points from a surface mesher
@throws IllegalArgumentException if there are less than 12 points (a tetrahedron)
static double[] getFerets(final List<List<Point3f>> surfacePoints) throws IllegalArgumentException {
    if (surfacePoints.size() < 12) {
        throw new IllegalArgumentException("Mesh has too few points");
    }
    ...
}
```

Reply:
> Empty surfaces (nPoints = 0) are already handled.

This exception checking finds a few surfaces with 0 points. Previously only `null` surfaces were handled.